src/rules.js

```diff
@@ -94,9 +94,23 @@ block.normal = merge({}, block);
 
 block.gfm = merge({}, block.normal, {
   nptable: /^ *([^|\n ].*\|.*)\n *([-:]+ *\|[-| :]*)(?:\n((?:.*[^>\n ].*(?:\n|$))*)\n*|$)/,
-  table: /^ *\|(.+)\n *\|?( *[-:]+[-| :]*)(?:\n((?: *[^>\n ].*(?:\n|$))*)\n*|$)/
+  table: '^ *\\|(.+)\\n' // Header
+    + ' *\\|?( *[-:]+[-| :]*)' // Align
+    + '(?:\\n((?:(?!^|>|\\n| |hr|heading|lheading|code|fences|list|html).*(?:\\n|$))*)\\n*|$)' // Cells
 });
 
+block.gfm.table = edit(block.gfm.table)
+  .replace('hr', block.hr)
+  .replace('heading', ' {0,3}#{1,6} +')
```

Comment:
```suggestion
  .replace('heading', ' {0,3}#{1,6} ')
```

No need to check for more than one space if one will match.

Reply:
This could also be fixed in the paragraph regex.

Reply:
Committed